sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -85,54 +90,75 @@ public void isSupportedAbsolute_returns_fileSystem_result_when_false() {
+    verify(fs).inputFiles(any());
   }
 
   @Test
-  public void getFileByRelativePath_when_no_indexed_files_returns_empty() {
-    FileSystem fs = createFileSystemForInputFiles(Collections.emptyList());
+  public void getFileByRelativePath_regex_test() {
+    // arrange
+    FileSystem fs = mock(FileSystem.class);
+    FilePredicates filePredicates = mock(FilePredicates.class);
+    when(fs.predicates()).thenReturn(filePredicates);
+    ArgumentCaptor<FilePredicate> captor = ArgumentCaptor.forClass(FilePredicate.class);
+    List<String> testInput = Arrays.asList(
+      "/_/some/path/file.cs",
+      "/_1/some/path/file.cs",
+      "/_10/some/path/file.cs",
+      "\\_2\\some\\path\\file.cs",
+      "\\_1234\\some\\path\\file.cs",
+      "\\_9999\\some/path/file.cs");
 
     // act
     ScannerFileService sut = new ScannerFileService("key", fs);
-    Optional<InputFile> result = sut.getFileByRelativePath("/_/some/path/file.cs");
+    testInput.forEach(sut::getFileByRelativePath);
 
     // assert
-    assertThat(result).isEmpty();
-    assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Found 0 indexed files for '/_/some/path/file.cs' (normalized to 'some/path/file.cs'). Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.");
+    verify(filePredicates, times(testInput.size())).and(any(), captor.capture());
+    List<RelativePathPredicate> predicates = captor.getAllValues().stream().map(obj -> (RelativePathPredicate) obj).collect(Collectors.toList());
+    for (RelativePathPredicate predicate : predicates) {
+      assertThat(predicate.getRelativePath()).isEqualTo("some/path/file.cs");
+    }
   }
 
   @Test
-  public void getFileByRelativePath_when_indexed_files_do_not_match_returns_empty() {
-    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("another/path/file.cs"), mockInput("some/file.cs")));
+  public void getFileByRelativePath_when_filesystem_returns_empty_returns_empty() {
+    FileSystem fs = createFileSystemForInputFiles(Collections.emptyList());
 
+    // act
     ScannerFileService sut = new ScannerFileService("key", fs);
     Optional<InputFile> result = sut.getFileByRelativePath("/_/some/path/file.cs");
 
+    // assert
     assertThat(result).isEmpty();
     assertThat(logTester.logs(LoggerLevel.DEBUG)).containsExactly("Found 0 indexed files for '/_/some/path/file.cs' (normalized to 'some/path/file.cs'). Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.");
   }
 
   @Test
-  public void getFileByRelativePath_when_multiple_indexed_files_match_returns_empty() {
-    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("root1/some/path/file.cs"), mockInput("root2/some/path/file.cs")));
+  public void getFileByRelativePath_when_multiple_indexed_files_match_returns_empty_and_logs() {
+    FileSystem fs = createFileSystemForInputFiles(Arrays.asList(mockInput("foo"), mockInput("bar")));
```

Comment:
The test is named `when_multiple_indexed_files_match`, but the link that the FS always returns all files is little hidden in this test. And it's not visible from `createFileSystemForInputFiles` name nor the `foo`/`bar` values. So it confuses me, because it loos like it's empty because none of those files matches.

Can we rename the method to something like `createFileSystemReturningAllFiles`?